src/data/list/basic.lean

```diff
@@ -2229,6 +2229,22 @@ begin
     apply hl _ (list.mem_cons_self _ _) }
 end
 
+lemma foldr_range_subset_of_range_subset {f : α → γ → γ} {g : β → γ → γ}
+  (hfg : set.range f ⊆ set.range g) (a) : set.range (foldr f a) ⊆ set.range (foldr g a) :=
```

Comment:
The `foldrl` version will be something like
```lean
  (hfg : set.range (flip f) ⊆ set.range (flip g)) (a) : set.range (foldl f a) ⊆ set.range (foldl g a) :=
```

Reply:
The proof of this variant doesn't seem to come immediately from the `foldr` variant. And since this is already a quite specific theorem I'm only really using for one purpose, I think it might be best just not to prove the `foldl` variant.

Reply:
I added a proof